deploy/src/main/deploy/helm/templates/hono-service-device-registry/hono-service-device-registry-deployment.yaml

```diff
@@ -1,5 +1,5 @@
 apiVersion: apps/v1beta1
-kind: Deployment
+kind: StatefulSet
```

Comment:
I see your point here and I agree that the stateful set model is a better fit for the example device registry in general. However, in practice the registry will only work with a single pod anyway, regardless of whether we use a *Deployment* or *StatefulSet*. Would you agree?

Reply:
True, but when you re-deploy/update k8s will boot up a second pod and this case there is a high probability to tun into a "Multi-Attach error" if the second pod goes onto another node in the cluster which can happen with `deployment`.

Reply:
I see, and with a *Stateful Set* k8s does what instead when re-deploying/updating?

Reply:
Well, in order to maintain its _sticky identity_ for pods it does not try do the update zero downtime, i.e. destroys the old pod first and then boots the new one.

I have not found a way to get that behaviour (i.e. no zero downtime) out of _deployment_. Based on all of the discussions around Multi-Attach error it seems _ StatefulSet_ is the solution of choice.

Reply:
Ok, I have no issues with changing to *Stateful Set* for the example registry here. We use it for demo purposes only anyway ...